frontend/src/Components/RoadDetails/ImageGallery.tsx

```diff
@@ -108,24 +132,28 @@ const ImageGallery: React.FC = () => {
           ))}
         </div>
       </div>
-      <button
-        className={`arrow-button left-arrow ${
-          scrollPosition === 0 ? 'disabled' : ''
-        }`}
-        onClick={() => handleScroll(-120)}
-        disabled={scrollPosition === 0}
-      >
-        &#9658;
-      </button>
-      <button
-        className={`arrow-button right-arrow ${
-          scrollPosition === (images.length - 1) * 124 ? 'disabled' : ''
-        }`}
-        onClick={() => handleScroll(120)}
-        disabled={scrollPosition === (images.length - 1) * 124}
-      >
-        &#9658;
-      </button>
+      {showArrows && (
+        <>
+          <button
+            className={`arrow-button left-arrow ${
+              scrollPosition === 0 ? 'disabled' : ''
+            }`}
+            onClick={() => handleScroll(-120)}
+            disabled={scrollPosition === 0}
+          >
+            &#9658;
+          </button>
```

Comment:
When doing:
- One to the right and then one to the left: everything is ok
- Two to the right and then I can only to one to the left (and I'm not able to go back to the first image)

Reply:
This has been fully fixed and the scrolling behavior feels way better. No bugs with the next branch update.